### DISCUSSION TITLE
Feature: Output all Next.js build errors concurrently.

### DISCUSSION THREAD

**Discussion by divinrkz** [vercel/next.js#32795] [Ideas]
### Describe the feature you'd like to request

Well, currently, Next.js build errors are emitted one by one. Let's say I have 2 build errors; one in `component1.js` and `component2.js`. When you run `next build` only one build error will be emitted to the cli. After a fix on the error in code, when you build again, you will find another unsolved error in other files which is stressing. For the worst case scenario, a codebase of over 100 files with possible unexpected build errors, emitting one by one is quite a heck.

I think it would be great to emit all build errors with a single build command.

### Describe the solution you'd like

Improve to compiler to build all files and emit errors concurrently instead of one by one.

### Describe alternatives you've considered

Addition writing all error logs to a file would also help.

**Comment by bsplosion** [comment]
Necroing in case you didn't see it in other threads: if you set an environment variable of `__NEXT_TEST_MODE=true`, there's undocumented behavior which enables gathering all errors during a `next build` call.  Not sure why it's internal-only and undocumented, but it's a _huge_ time saver.

Next.js should really just make it a first-class flag and document, this is a very common need when running a build after a big batch of changes which have only been applied during `next dev`.
